Applications/Utils/ModelPreparation/PartitionMesh/NodeWiseMeshPartitioner.cpp

```diff
@@ -346,11 +436,50 @@ void NodeWiseMeshPartitioner::renumberNodeIndices(
     }
 }
 
-void NodeWiseMeshPartitioner::writeNodePropertiesBinary(
-    const std::string& file_name_base) const
+template <typename T>
+void writePropertyVectorValuesBinary(std::ostream& os,
+                                     MeshLib::PropertyVector<T> const& pv)
 {
-    auto const& property_names(_partitioned_properties.getPropertyVectorNames(
-        MeshLib::MeshItemType::Node));
+    std::size_t number_of_components(pv.getNumberOfComponents());
+    std::size_t number_of_tuples(pv.getNumberOfTuples());
+    std::vector<T> property_vector_buffer;
+    property_vector_buffer.resize(number_of_tuples * number_of_components);
+    for (std::size_t i = 0; i < pv.getNumberOfTuples(); ++i)
+    {
+        for (std::size_t c(0); c < number_of_components; ++c)
+            property_vector_buffer[i * number_of_components + c] =
+                pv.getComponent(i, c);
+    }
+    os.write(reinterpret_cast<char*>(property_vector_buffer.data()),
```

Comment:
Can't `pv.data()` (`pv` is a vector, isn't it?) be used directly? Or is the order different?

Reply:
Nice! Didn't see this one. Including...